types/recompose/index.d.ts

```diff
@@ -151,7 +151,7 @@ declare module 'recompose' {
       [updaterName: string]: StateHandler<TState>;
     };
     type StateUpdaters<TOutter, TState, TUpdaters> = {
-      [updaterName in keyof TUpdaters]: (state: TState, props: TOutter) => StateHandler<TState>;
+      [updaterName in keyof TUpdaters]: (state: TState, props: TOutter) => TUpdaters[updaterName];
     };
     export function withStateHandlers<TState, TUpdaters extends StateHandlerMap<TState>, TOutter = {}>(
```

Comment:
BTW don't you think it would be even better to replace `TUpdaters extends StateHandlerMap<TState>` with `TUpdaters = StateHandlerMap<TState>`, so we could set a custom `TUpdaters` without extending the default `StateHandlerMap` ? It shouldn't introduce any breaking change.